test/NomoPlayersDropMechanic.test.ts

```diff
@@ -79,24 +79,143 @@ describe("NomoPlayersDropMechanic tests", function () {
+    const erc20Address = await nomoPlayersDropMechanicContract.connect(deployer).erc20Address();
+    expect(testAddress).to.equal(erc20Address)
+  });
+
+  it("must fail to set ERC20 address if msg.sender isn't owner", async function () {
+    await expect(nomoPlayersDropMechanicContract.connect(user).setERC20Address(testAddress))
+      .to.be.revertedWith("Ownable: caller is not the owner");
+  });
+
+  it("must fail to set ERC20 address isn't valid", async function () {
+    await expect(nomoPlayersDropMechanicContract.connect(deployer).setERC20Address(zeroAddress))
+      .to.be.revertedWith("Not a valid address!");
+  });
+  
+  
+  it("should emit LogStrategyContractAddressSet event", async function () {
+    await expect(nomoPlayersDropMechanicContract.connect(deployer).setStrategyContractAddress(testAddress)).to.emit(nomoPlayersDropMechanicContract, "LogStrategyContractAddressSet");
+  });
+
+  it("should set Strategy contract address", async function () {
+    await nomoPlayersDropMechanicContract.connect(deployer).setStrategyContractAddress(testAddress);
+    const strategyContractAddress = await nomoPlayersDropMechanicContract.connect(deployer).strategyContractAddress();
+    expect(testAddress).to.equal(strategyContractAddress)
+  });
+
+  it("must fail to set Strategy contract address if msg.sender isn't owner", async function () {
+    await expect(nomoPlayersDropMechanicContract.connect(user).setStrategyContractAddress(testAddress))
+      .to.be.revertedWith("Ownable: caller is not the owner");
+  });
+
+  it("must fail to set Strategy contract address isn't valid", async function () {
+    await expect(nomoPlayersDropMechanicContract.connect(deployer).setStrategyContractAddress(zeroAddress))
+      .to.be.revertedWith("Not a valid address!");
+  });
+  
+  
+  it("should emit LogDaoWalletAddressSet event", async function () {
+    await expect(nomoPlayersDropMechanicContract.connect(deployer).setDaoWalletAddress(testAddress)).to.emit(nomoPlayersDropMechanicContract, "LogDaoWalletAddressSet");
+  });
+
+  it("should set Dao wallet address", async function () {
+    await nomoPlayersDropMechanicContract.connect(deployer).setDaoWalletAddress(testAddress);
+    const daoWalletAddress = await nomoPlayersDropMechanicContract.connect(deployer).daoWalletAddress();
+    expect(testAddress).to.equal(daoWalletAddress)
+  });
+
+  it("must fail to set Dao wallet address if msg.sender isn't owner", async function () {
+    await expect(nomoPlayersDropMechanicContract.connect(user).setDaoWalletAddress(testAddress))
+      .to.be.revertedWith("Ownable: caller is not the owner");
+  });
+
+  it("must fail to set Dao wallet address isn't valid", async function () {
+    await expect(nomoPlayersDropMechanicContract.connect(deployer).setDaoWalletAddress(zeroAddress))
+      .to.be.revertedWith("Not a valid address!");
+  });
+  
+  
+  it("should set presale start date", async function () {
```

Comment:
```suggestion
it("should set presale start date", async function () {

    const blockNumber = await ethers.provider.getBlockNumber()
    const block = await ethers.provider.getBlock(blockNumber)
    const timestamp = block.timestamp
    const TWO_DAYS = 60 * 60 * 24 * 2;


    const unixTimeStampStartDate = timestamp + TWO_DAYS;
    await nomoPlayersDropMechanicContract.connect(deployer).setPresaleStartDate(unixTimeStampStartDate);
    const presaleStartDate = await nomoPlayersDropMechanicContract.connect(deployer).presaleStartDate();
    expect(unixTimeStampStartDate).to.equal(presaleStartDate)
  });
```

Another thing you should consider is to export getting the `block timestamp` as a function as you are to rephrase one more test if i remember correctly